#! /usr/bin/env python

PACKAGE='seabee3_driver_base'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#Name            Type   Reconfiguration level             Description         Default Min Max
gen.add( "is_killed",         bool_t,   SensorLevels.RECONFIGURE_RUNNING, "kill switch", True )
gen.add( "internal_pressure", int_t,    SensorLevels.RECONFIGURE_RUNNING, "internal pressure",   3000,  0,   5000 )
gen.add( "external_pressure", int_t,    SensorLevels.RECONFIGURE_RUNNING, "external pressure",   890,   0,   5000 )
gen.add( "override_imu",      bool_t,   SensorLevels.RECONFIGURE_RUNNING, "override IMU inputs", False )
gen.add( "roll",              double_t, SensorLevels.RECONFIGURE_RUNNING, "roll",                0.0,   -360.0, 360.0 )
gen.add( "pitch",             double_t, SensorLevels.RECONFIGURE_RUNNING, "pitch",               0.0,   -360.0, 360.0 )
gen.add( "yaw",               double_t, SensorLevels.RECONFIGURE_RUNNING, "yaw",                 0.0,   -360.0, 360.0 )

exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "FakeSeabee"))

